cmd/mycelo/main.go

```diff
@@ -422,19 +442,25 @@ func validatorRun(ctx *cli.Context) error {
 
 func loadBot(ctx *cli.Context) error {
 	env, err := readEnv(ctx)
+	if ctx.IsSet("tps") {
+		env.Config.LoadTestTPS = ctx.Int("tps")
+	}
 	if err != nil {
 		return err
 	}
 
 	runCtx := context.Background()
 
 	// TODO: Pull all of these values from env.json
+	client, err := ethclient.Dial(env.IPC())
+	if err != nil {
+		return err
+	}
 	return loadbot.Start(runCtx, &loadbot.Config{
-		Accounts:         env.DeveloperAccounts(),
-		Amount:           big.NewInt(10000000),
-		TransactionDelay: 1 * time.Second,
-		ClientFactory: func() (*ethclient.Client, error) {
-			return ethclient.Dial("http://localhost:8545")
-		},
+		Accounts:              env.DeveloperAccounts(),
+		Amount:                big.NewInt(10000000),
+		TransactionsPerSecond: env.Config.LoadTestTPS,
+		Clients:               []*ethclient.Client{client},
```

Comment:
something i'm a bit dubious here is that i don't know if the ipc client connection is able to handle several concurrent requests; and if instead it just sequence requests.

RPC is a request -> response protocol. So if have 100 goroutines using the SAME ipc connection; are requests queued one after the other, or do they happen concurrently.

I suspect they are queued, and hence you need to wait for the first request to finish to process the second request.

That's why initially, i used multiple http connections, since every connection was independente, they can happen concurrently, and then it's a question on how many requests can the rpc server handle concurrently.

Reply:
The IPC client has been significantly more stable that HTTP connections (even when I would play with the number of connections). If you go 1:1 with HTTP connections to dev accounts, you can easily saturate the RPC server with too many connections (and you get `err connection refused`). Even limiting it to 50 or 100 HTTP clients, it'd run fine for a bit, and then eventually have an issue and be unable to send the transaction.

Reply:
ok, a small change then is to use all the validators ipc connections. so you have a subset of clients connected to each validator

Reply:
Yea, I can use all validator connections. During my tests I actually only run it with one validator, but that makes more sense.